### ISSUE TITLE
Arrow keys don't move widgets after initial creation

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#2928]
With M5 and M6, in absolute positioning mode (i.e., sketch mode), immediately after dropping a widget onto the page, the user can use the arrow keys to move the widget up/down/left/right. But with M7, arrows keys only work after the user clicks on the newly dropped widget.

I'm 99% sure this is a focus issue. With M5 and M6, after adding a new widget, I included code that put the focus into the user's iframe just so that the arrow keys would work immediately. I'm not sure what changed to cause this "force focus into user iframe" logic to no longer work.

**Comment by doronrosenberg** [comment]
@JonFerraiolo roughly where would the code be that did the focusing?

**Comment by JonFerraiolo** [comment]
I'm pretty sure that the user document got the focus in M6 due to this line at CreateTool.js:413:

```
            userdoc.defaultView.focus();    // Make sure the userdoc is the focus object for keyboard events
```

which is in `_cleanupActions()`, which is called at the very end of the onMouseUp handler in CreateTool.js.

**Comment by JonFerraiolo** [comment]
I am hoping that this fix will also fix the Esc key problem with the polyline create tool described in #2926

**Comment by doronrosenberg** [comment]
This isn't a focus issue it seems, the reason is:

```
 if(event && this._moverWidget){
```

there is no moverWidget when you dnd.

**Comment by doronrosenberg** [comment]
@JonFerraiolo 

To be more exact, SelectTool::onMouseDown handles the moverWidget logic, and that doesn't get called when you just create and select a widget.

**Comment by doronrosenberg** [comment]
@JonFerraiolo 

So commit 0dc74f6abe1068391cab4cdfe8fea0de6ff966ab is what seems to have broken this.  There were no checks for `this._moverWidget` in the key listeners.  I could change the code to allow the keyboard arrow events to work if there is no `this._moverWidget` but I don't want to potentially break anything since I don't know much about this code.

**Comment by JonFerraiolo** [comment]
I don't think it was 0dc74f6 because that was committed before Preview6 shipped and Preview6 doesn't have this problem.

I'll look into this some more...

**Comment by JonFerraiolo** [comment]
OK, I figured it out, and I have a fix. @doronrosenberg was correct in saying that  0dc74f6 broke the feature. But the reason it seemed to work in Preview6 when I tested there was the result of two bugs, cancelling themselves out. First bug was this line as pointed out by @doronrosenberg:

```
 if(event && this._moverWidget){
```

Second bug was that this._moverWidget wasn't getting cleared when the move operation failed. Therefore, the arrow keys failed with the first widget, but worked with subsequent widgets. (Same thing both in Preview6 and in latest code).

Fix coming momentarily.
